### PR TITLE
[jws] Add ability to verify payload using "jku"

### LINKED ISSUE #512: [jws] Automatically respect JWKS in `jku` to verify the JWS payload
Probably something like `jws.Verify(..., jws.WithAutoFetchJKU(true))`.

Because this would be fetching the remote resource, we must be aware of security impact.
We probably should document it is highly recommended to ONLY use this option with a whitelist
(also, we would have to create wrapper options? to pass `jwk.FetchOption` to `jws.Verify()`)

### PR DESCRIPTION
**Pull request by lestrrat**
fixes #512

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -5,6 +5,17 @@ v1.2.14 UNRELEASED
 [New Features]
   * `jwk.Fetch()` and `(*jwk.AutoRefresh).Configure()` can now take `jwk.Whitelist`
     object to check for the validity of a url to be fetched
+  * `jws.VerifyAuto()` has been added to verify payloads that can be verified
+    using the JWK set provided in the "jku" field. This function is purposely
+    separated from the `jws.Verify()` function because 1) the required parameters
+    are different, and 2) Users MUST be aware that they are doing a totally
+    different operation than a regular `jws.Verify()`
+
+[Buf fixes]
```
Comment (gonzaloserrano) [inline Changes:14]:
```suggestion
[Bug fixes]
```

Comment (lestrrat) [reply Changes:14]: oops, thanks! :D

File: jws/jws.go [hunk 2]
```diff
@@ -312,12 +365,37 @@ func verifyJSON(signed []byte, alg jwa.SignatureAlgorithm, key interface{}, dst
...
+			}
+
+			verifier, err := NewVerifier(ctx.alg)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to create verifier")
+			}
+
+			if _, err := ctx.tryVerify(verifier, sig.protected, buf.Bytes(), sig.signature, m.payload); err == nil {
+				if ctx.dst != nil {
+					*(ctx.dst) = m
```
Comment (gonzaloserrano) [inline jws/jws.go:384]: not that I have any knowledge about how this works but I noticed this line and L393 are not covered by tests

Comment (lestrrat) [reply jws/jws.go:384]: Thanks, addressed in 7a2ff0c

**Pull request merged**